Tasks/Common/nuget-task-common/LocationHelpers.ts

```diff
@@ -0,0 +1,130 @@
+    var serviceIdUppercase = serviceId.toUpperCase();
+    var serviceDefinitions = connectionData.locationServiceData.serviceDefinitions;
+    return serviceDefinitions.find(service => service.identifier.toUpperCase() == serviceIdUppercase);
+}
+
+function hasServicesOfType(connectionData: locationApi.ConnectionData, serviceType: string): boolean {
+    var serviceTypeUpppercase = serviceType.toUpperCase();
+    var serviceDefinitions = connectionData.locationServiceData.serviceDefinitions;
+    return serviceDefinitions.some(service => service.serviceType.toUpperCase() == serviceTypeUpppercase);
+}
+
+export function getIdentityDisplayName(identity: locationApi.Identity): string {
+    if (identity.customDisplayName) {
+        return identity.customDisplayName;
+    }
+    if (identity.providerDisplayName) {
+        return identity.providerDisplayName;
+    }
+
+    return null;
+}
+
+export function getIdentityAccount(identity: locationApi.Identity): string {
+    if (identity.properties['Account']) {
+        return identity.properties['Account'].$value
+    }
+
+    return null;
+}
+
+export function getAllAccessMappingUris(connectionData: locationApi.ConnectionData): string[] {
+    var accessMappings = connectionData.locationServiceData.accessMappings;
+    return accessMappings.map(getUriForAccessMapping);
+}
+
+export function getConnectionDataForArea(serviceUri: string, areaName: string, areaId: string, accessToken: string): Q.Promise<locationApi.ConnectionData> {
+    return getConnectionData(serviceUri, accessToken)
+        .then(connectionData => {
+            tl.debug("successfully loaded origin service location data");
+            if (hasServicesOfType(connectionData, areaName)) {
+                tl.debug(util.format("found %s routes on %s", areaName, serviceUri));
+                return Q(connectionData);
+            }
+            else {
+                tl.debug(util.format("did not find %s routes directly on %s, trying SPS", areaName, serviceUri));
+                const rootLocationServiceId = "951917AC-A960-4999-8464-E3F0AA25B381";
+                var sps = findServiceByIdentifier(connectionData, rootLocationServiceId);
+                if (!sps) {
+                    throw new Error("The current service does not support NuGet and does not have SPS");
+                }
+
+                var spsUri = getUriForServiceDefinition(sps);
+                tl.debug(util.format("found SPS at %s", spsUri));
+
+                return getConnectionData(spsUri, accessToken)
+                    .then(spsConnectionData => {
+                        tl.debug("successfully loaded SPS location data");
+                        var areaService = findServiceByIdentifier(spsConnectionData, areaId);
+                        if (!areaService) {
+                            var error = new Error("Could not locate nuget service in the root location service");
```

Comment:
I don't see anything in the code that's nuget specific. So maybe change the string?